quickwit/quickwit-indexing/src/split_store/indexing_split_cache.rs

```diff
@@ -170,14 +227,39 @@ impl InnerLocalSplitStore {
         split_id: Ulid,
         to_folder: &Path,
     ) -> StorageResult<Option<PathBuf>> {
-        let Some(split_folder) = self.split_folders.remove(&split_id) else {
+        if !self.split_registry.remove(split_id) {
+            // The split is simply not in cache.
             return Ok(None);
         };
         let from_path = self.split_path(split_id);
         let to_full_path = to_folder.join(from_path.file_name().unwrap());
-        tokio::fs::rename(&from_path, &to_full_path).await?;
-        self.split_store_space_quota
-            .remove_split(split_folder.num_bytes);
+
```

Comment:
at this point in the method the split registry and the directory content is inconsistent.